### PR TITLE
Fix resource listing

### PR DESCRIPTION
**Pull request by thirdeyenick**
This fixes the not working resource listing over all projects. We now search for the users projects and search in them for the corresponding resource if `--all-projects/-A` was given. As this required quite some changes in test setups (projects have to be around when testing the `--all-projects` functionality), this PR also extends the internal test framework which can be used to setup a fake clientset. All manual creations of fake clientsets have been replaced by the test framework.

The main changes of this PR are in `internal/test/client.go` and `get/get.go`. All the other changes are test framework adaptions or cleanups of existing tests.

### CODE REVIEW

**Review by ctrox — APPROVED** [review]
One negative side-effect that I noticed is that now Niners with full permissions can't list objects in all namespaces anymore. But I think we can't really have both so we might just need to live with it.

**Comment by thirdeyenick** [comment]
> One negative side-effect that I noticed is that now Niners with full permissions can't list objects in all namespaces anymore. But I think we can't really have both so we might just need to live with it.

Yeah, true this a negative side effect. This could be changed by adding an option, but I'd say it is rather uncommon to have something like an 'admin' option in such a tool.

File: api/config/extension_config.go [hunk 1]
```diff
@@ -1,54 +1,43 @@
...
-		err,
-		format.Command().Login(),
-	)
-}
-
-// Config is used to store information in the kubeconfig context created
-type Config struct {
+// ExtensionConfig is used to store information in the kubeconfig context
+// created
+type ExtensionConfig struct {
```
Comment (ctrox) [inline api/config/extension_config.go:40]: Nit: the name `config.ExtensionConfig` stutters a bit.

Comment (thirdeyenick) [reply api/config/extension_config.go]: renamed it to `config.Extension` and adapted the other related functions

**Pull request merged**